src/protocols/secure_channel/tests/TestPASESession.cpp

```diff
@@ -26,13 +26,44 @@
 
 #include <core/CHIPCore.h>
 #include <core/CHIPSafeCasts.h>
+#include <messaging/tests/MessagingContext.h>
 #include <protocols/secure_channel/PASESession.h>
 #include <stdarg.h>
 #include <support/CHIPMem.h>
 #include <support/CodeUtils.h>
 #include <support/UnitTestRegistration.h>
 
 using namespace chip;
+using namespace chip::Inet;
+using namespace chip::Transport;
+using namespace chip::Messaging;
+using namespace chip::Protocols;
+
+using TestContext = chip::Test::MessagingContext;
+
+class LoopbackTransport : public Transport::Base
+{
+public:
+    /// Transports are required to have a constructor that takes exactly one argument
+    CHIP_ERROR Init(const char * unused) { return CHIP_NO_ERROR; }
+
+    CHIP_ERROR SendMessage(const PacketHeader & header, const PeerAddress & address, System::PacketBufferHandle msgBuf) override
+    {
+        ReturnErrorOnFailure(mMessageSendError);
+        mNumMessageSend++;
+        HandleMessageReceived(header, address, std::move(msgBuf));
+
+        return CHIP_NO_ERROR;
+    }
+
+    bool CanSendToPeer(const PeerAddress & address) override { return true; }
+
+    uint32_t mNumMessageSend     = 0;
```

Comment:
```suggestion
    uint32_t mMessagesSent     = 0;
```
?  Or `mSentMessageCount`?